### PR TITLE
SE-1617 update ecommerce docs

### PR DESCRIPTION
**Pull request by samuelallan72**
This updates the e-commerce installation docs for ocim.

**Test instructions**:

- Follow conversation on [SE-1617](https://tasks.opencraft.com/browse/SE-1617) and verify that updates here match what was done and the config on the client's instance.
- check that the markdown renders in github nicely.

**Reviewer**:

- [x] @pomegranited

### CODE REVIEW

**Review by pomegranited — CHANGES_REQUESTED** [review]

File: documentation/howtos.md [hunk 1]
```diff
@@ -4,58 +4,125 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
    Ensure the staff user has a user profile associated (i.e. set a Full Name).
-1. In Django Admin > OAuth2 > Clients, there should already be clients created
-   for ecommerce and discovery.
+1. In Django Admin > OAuth2 > Clients (`/admin/oauth2/client/`), there should
+   already be clients created for ecommerce and discovery.
    If not, [create and register new clients](http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/ecommerce/install_ecommerce.html#configure-edx-openid-connect-oidc)
    for each service.  You'll need the client IDs and client secrets for the next
-   steps.
-   Ensure that both clients are attached to the staff user updated above.
+   steps.  Ensure that both clients are attached to the staff user updated above.
```
Comment (pomegranited) [inline documentation/howtos.md]: The `ecommerce_worker` user isn't a staff user, which might be related to the permissions issue I'm having with creating course seats in ecommerce?  I tried elevating the user and it didn't fix it though.  Also, I think reprovisioning might re-set this back to a non-staff user.

File: documentation/howtos.md [hunk 2]
```diff
@@ -4,58 +4,125 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
+   ```
+   sudo -u discovery -Hs
+   cd
+   source discovery_env
+   source venvs/discovery/bin/activate
+   cd discovery
+   # TODO: what should --site-domain be here?
+   python manage.py create_or_update_partner \
+     --site-id 1 \
+     --site-domain 'external.lms.domain' \
```
Comment (pomegranited) [inline documentation/howtos.md]: Should be the discovery site domain, `discovery.external.lms.domain`

File: documentation/howtos.md [hunk 3]
```diff
@@ -4,58 +4,125 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
+     --code 'partn_id' \
+     --name 'Montessori' \
```
Comment (pomegranited) [inline documentation/howtos.md]: Change to 'client name'

File: documentation/howtos.md [hunk 4]
```diff
@@ -4,58 +4,125 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
-        https://discovery-<your-instance>.opencraft.hosting/login
+        https://ecommerce.<your-instance>.opencraft.hosting/login
+        https://discovery.<your-instance>.opencraft.hosting/login
 
 1. Verify that the discovery cronjob runs without errors:
 
-        sudo -s -c . /edx/app/discovery/discovery_env; /edx/bin/manage.discovery refresh_course_metadata
+        sudo -u discovery -Hs
+        source /edx/app/discovery/discovery_env
+        /edx/bin/manage.discovery refresh_course_metadata
```
Comment (pomegranited) [inline documentation/howtos.md]: Worth referencing the e-commerce usage docs here too: https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/ecommerce/create_products/index.html

File: documentation/howtos.md (hunk 1 above)
Comment (swalladge) [reply documentation/howtos.md]: The issue was that some permissions were missing for ecommerce_worker. Elevating to staff doesn't override permissions granted; I think only superuser does that.

Comment (pomegranited) [reply documentation/howtos.md]: Huh interesting.. Thanks for [adding those](https://github.com/open-craft/opencraft/pull/523/files#diff-e62c375eb49e96b787ac278c5d78e70dR76)!

File: documentation/howtos.md [hunk 5]
```diff
@@ -4,61 +4,153 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
+
+TODO: add more complete manual test checklist
```
Comment (pomegranited) [inline documentation/howtos.md]: 1. Create a new course to test ecommerce.
1. Add a new [Course seat](https://edx-ecommerce.readthedocs.io/en/latest/create_products/create_course_seats.html) for this course, with a nominal charge (e.g. $1).
1. Optionally add a [Coupon](https://edx-ecommerce.readthedocs.io/en/latest/create_products/create_coupons.html) for the course, if the client will use coupons.
1. Enroll a new user into the test course, and verify that the coupon works, payments are processed, and the enrollment succeeds.
   Ideally, the payment processor will be initially configured as a sandbox service for testing, and so you can use test credit card numbers as provided by the payment processor.

**Review by pomegranited — CHANGES_REQUESTED** [review]

File: documentation/howtos.md (hunk 5 above)
Comment (swalladge) [reply documentation/howtos.md]: Awesome, thanks! :D

File: documentation/howtos.md [hunk 6]
```diff
@@ -4,61 +4,153 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
+
 Once the spawn is complete, you'll need to take the following steps to finish setup
 (these are one-time actions that are stored in the database):
 
 1. Create/choose a staff user to use for the OAuth2 clients.
    Ensure the staff user has a user profile associated (i.e. set a Full Name).
-1. In Django Admin > OAuth2 > Clients, there should already be clients created
-   for ecommerce and discovery.
+   (Note that users may be auto created in provision process and this step thus
+   not required; check the next step before creating users.)
```
[the remark this answers is not in the document]
Comment (swalladge) [reply documentation/howtos.md]: Ah yes, that is much nicer, thanks!

File: documentation/howtos.md [hunk 7]
```diff
@@ -4,61 +4,164 @@ By default, instances are not provisioned with either ecommerce or the Course Di
...
+ECOMMERCE_PAYMENT_PROCESSOR_CONFIG:
+    partn_id: # this is arbitrary, limited to 8 characters; will be used later
+      paypal:
+        ...
+
+# Have to build the full broker URLs to ensure the vhost path is included
+ECOMMERCE_BROKER_URL: '{{ EDXAPP_CELERY_BROKER_TRANSPORT }}://{{ EDXAPP_CELERY_USER }}:{{ EDXAPP_CELERY_PASSWORD }}@{{ EDXAPP_CELERY_BROKER_HOSTNAME }}{{ EDXAPP_CELERY_BROKER_VHOST }}'
+ECOMMERCE_WORKER_BROKER_URL: '{{ EDXAPP_CELERY_BROKER_TRANSPORT }}://{{ EDXAPP_CELERY_USER }}:{{ EDXAPP_CELERY_PASSWORD }}@{{ EDXAPP_CELERY_BROKER_HOSTNAME }}{{ EDXAPP_CELERY_BROKER_VHOST }}'
+ECOMMERCE_WORKER_ECOMMERCE_API_ROOT: '{{ ECOMMERCE_ECOMMERCE_URL_ROOT }}/api/v2/'
+```
```
Comment (pomegranited) [inline documentation/howtos.md]: Changing the currency was pretty involved, so the details of that should be noted here.

> To change the default currency used in the ecommerce system from USD ($) to, for example, British points (£), add this extra configuration:

```
# Use a configuration branch which contains this fix:
# https://github.com/open-craft/configuration/pull/119/files
EDXAPP_PAID_COURSE_REGISTRATION_CURRENCY: ['gbp', '£']
ECOMMERCE_OSCAR_DEFAULT_CURRENCY: 'GBP'
ECOMMERCE_REPOS:
  - PROTOCOL: '{{ COMMON_GIT_PROTOCOL }}'
    DOMAIN: '{{ COMMON_GIT_MIRROR }}'
    PATH: 'open-craft'
    REPO: 'ecommerce.git'
    # Use a branch which contains this fix:
    # https://github.com/open-craft/ecommerce/pull/13/files
    VERSION: 'jill/set-default-currency'  # FIXME: change to opencraft-release/ironwood.2
    DESTINATION: "{{ ecommerce_code_dir }}"
    SSH_KEY: '{{ ECOMMERCE_GIT_IDENTITY }}'
```

**Review by pomegranited — APPROVED** [review]
Thanks for making these changes @swalladge !  I've requested that the [details about the currency fix](https://github.com/open-craft/opencraft/pull/523/files#diff-e62c375eb49e96b787ac278c5d78e70dR45) be added to this PR as well, but this is good to merge once that's done.

:+1:

- [x] I tested this by checking the instance configuration for our new client using ecommerce, and by testing course purchases on that instance.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> docs changes only.
- [x] Updates documentation
- [x] <strike>I made sure any change in configuration variables is reflected in the corresponding
    client's instance configuration.</strike> N/A

**Comment by pomegranited** [comment]
👍  to the added (though unfortunate) hack required to get the course discovery indexing to play nice.  So glad we nutted this out!

Great addition to our docs, @swalladge !

**Pull request merged**